File: 1539-kth-missing-positive-number/1539-kth-missing-positive-number.py
class Solution(object):
    def findKthPositive(self, arr, k):
        """
        :type arr: List[int]
        :type k: int
        :rtype: int
        """
        
        
        l, r = 0, len(arr) - 1
        
        while l <= r:
            mid = (l+r)//2
            
            missing = arr[mid] - (mid + 1)
            
            if missing < k:
                l = mid + 1
            else:
                r = mid - 1
        
        return r + 1 + k

File: 1539-kth-missing-positive-number/1539-kth-missing-positive-number.java
class Solution {
    public int findKthPositive(int[] arr, int k) {
        int l = 0, r = arr.length - 1;
        
        while (l <= r){
            int mid = l + (r - l) / 2;
            int missingNums = arr[mid] - (mid + 1);
            
            if (missingNums < k){
                l = mid + 1;
            }
            else r = mid - 1;
        }
        return r + k + 1;
    }
}